From 9c18546df3e176b4462813490350f02eece09def Mon Sep 17 00:00:00 2001
From: Hien Huynh <hien.huynh.px@renesas.com>
Date: Tue, 28 Jul 2020 15:02:18 +0700
Subject: [PATCH 409/458] arm64: dts: renesas: r8a774a3: add VIN and CSI-2
 nodes

This patch adds VIN{0,1,2,3,4,5,6,7} and CSI{CSI40, CSI20} device node
support for r8a774a3 SoC.

Signed-off-by: Hien Huynh <hien.huynh.px@renesas.com>
---
 arch/arm64/boot/dts/renesas/r8a774a3.dtsi | 110 +++++++++++++++++++++++++++---
 1 file changed, 100 insertions(+), 10 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/r8a774a3.dtsi b/arch/arm64/boot/dts/renesas/r8a774a3.dtsi
index 4677311..a90e69d 100644
--- a/arch/arm64/boot/dts/renesas/r8a774a3.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a774a3.dtsi
@@ -2089,7 +2089,14 @@
 		};
 
 		vin0: video@e6ef0000 {
+			compatible = "renesas,vin-r8a774a3";
 			reg = <0 0xe6ef0000 0 0x1000>;
+			interrupts = <GIC_SPI 188 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 811>;
+			power-domains = <&sysc R8A774A3_PD_ALWAYS_ON>;
+			resets = <&cpg 811>;
+			renesas,id = <0>;
+			status = "disabled";
 
 			ports {
 				#address-cells = <1>;
@@ -2103,17 +2110,25 @@
 
 					vin0csi20: endpoint@0 {
 						reg = <0>;
+						remote-endpoint= <&csi20vin0>;
 					};
 					vin0csi40: endpoint@2 {
 						reg = <2>;
+						remote-endpoint= <&csi40vin0>;
 					};
 				};
 			};
-			/* placeholder */
 		};
 
 		vin1: video@e6ef1000 {
+			compatible = "renesas,vin-r8a774a3";
 			reg = <0 0xe6ef1000 0 0x1000>;
+			interrupts = <GIC_SPI 189 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 810>;
+			power-domains = <&sysc R8A774A3_PD_ALWAYS_ON>;
+			resets = <&cpg 810>;
+			renesas,id = <1>;
+			status = "disabled";
 
 			ports {
 				#address-cells = <1>;
@@ -2127,17 +2142,25 @@
 
 					vin1csi20: endpoint@0 {
 						reg = <0>;
+						remote-endpoint= <&csi20vin1>;
 					};
 					vin1csi40: endpoint@2 {
 						reg = <2>;
+						remote-endpoint= <&csi40vin1>;
 					};
 				};
 			};
-			/* placeholder */
 		};
 
 		vin2: video@e6ef2000 {
+			compatible = "renesas,vin-r8a774a3";
 			reg = <0 0xe6ef2000 0 0x1000>;
+			interrupts = <GIC_SPI 190 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 809>;
+			power-domains = <&sysc R8A774A3_PD_ALWAYS_ON>;
+			resets = <&cpg 809>;
+			renesas,id = <2>;
+			status = "disabled";
 
 			ports {
 				#address-cells = <1>;
@@ -2151,17 +2174,25 @@
 
 					vin2csi20: endpoint@0 {
 						reg = <0>;
+						remote-endpoint= <&csi20vin2>;
 					};
 					vin2csi40: endpoint@2 {
 						reg = <2>;
+						remote-endpoint= <&csi40vin2>;
 					};
 				};
 			};
-			/* placeholder */
 		};
 
 		vin3: video@e6ef3000 {
+			compatible = "renesas,vin-r8a774a3";
 			reg = <0 0xe6ef3000 0 0x1000>;
+			interrupts = <GIC_SPI 191 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 808>;
+			power-domains = <&sysc R8A774A3_PD_ALWAYS_ON>;
+			resets = <&cpg 808>;
+			renesas,id = <3>;
+			status = "disabled";
 
 			ports {
 				#address-cells = <1>;
@@ -2175,17 +2206,25 @@
 
 					vin3csi20: endpoint@0 {
 						reg = <0>;
+						remote-endpoint= <&csi20vin3>;
 					};
 					vin3csi40: endpoint@2 {
 						reg = <2>;
+						remote-endpoint= <&csi40vin3>;
 					};
 				};
 			};
-			/* placeholder */
 		};
 
 		vin4: video@e6ef4000 {
+			compatible = "renesas,vin-r8a774a3";
 			reg = <0 0xe6ef4000 0 0x1000>;
+			interrupts = <GIC_SPI 174 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 807>;
+			power-domains = <&sysc R8A774A3_PD_ALWAYS_ON>;
+			resets = <&cpg 807>;
+			renesas,id = <4>;
+			status = "disabled";
 
 			ports {
 				#address-cells = <1>;
@@ -2199,17 +2238,25 @@
 
 					vin4csi20: endpoint@0 {
 						reg = <0>;
+						remote-endpoint= <&csi20vin4>;
 					};
 					vin4csi40: endpoint@2 {
 						reg = <2>;
+						remote-endpoint= <&csi40vin4>;
 					};
 				};
 			};
-			/* placeholder */
 		};
 
 		vin5: video@e6ef5000 {
+			compatible = "renesas,vin-r8a774a3";
 			reg = <0 0xe6ef5000 0 0x1000>;
+			interrupts = <GIC_SPI 175 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 806>;
+			power-domains = <&sysc R8A774A3_PD_ALWAYS_ON>;
+			resets = <&cpg 806>;
+			renesas,id = <5>;
+			status = "disabled";
 
 			ports {
 				#address-cells = <1>;
@@ -2223,17 +2270,25 @@
 
 					vin5csi20: endpoint@0 {
 						reg = <0>;
+						remote-endpoint= <&csi20vin5>;
 					};
 					vin5csi40: endpoint@2 {
 						reg = <2>;
+						remote-endpoint= <&csi40vin5>;
 					};
 				};
 			};
-			/* placeholder */
 		};
 
 		vin6: video@e6ef6000 {
+			compatible = "renesas,vin-r8a774a3";
 			reg = <0 0xe6ef6000 0 0x1000>;
+			interrupts = <GIC_SPI 176 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 805>;
+			power-domains = <&sysc R8A774A3_PD_ALWAYS_ON>;
+			resets = <&cpg 805>;
+			renesas,id = <6>;
+			status = "disabled";
 
 			ports {
 				#address-cells = <1>;
@@ -2247,17 +2302,25 @@
 
 					vin6csi20: endpoint@0 {
 						reg = <0>;
+						remote-endpoint= <&csi20vin6>;
 					};
 					vin6csi40: endpoint@2 {
 						reg = <2>;
+						remote-endpoint= <&csi40vin6>;
 					};
 				};
 			};
-			/* placeholder */
 		};
 
 		vin7: video@e6ef7000 {
+			compatible = "renesas,vin-r8a774a3";
 			reg = <0 0xe6ef7000 0 0x1000>;
+			interrupts = <GIC_SPI 171 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 804>;
+			power-domains = <&sysc R8A774A3_PD_ALWAYS_ON>;
+			resets = <&cpg 804>;
+			renesas,id = <7>;
+			status = "disabled";
 
 			ports {
 				#address-cells = <1>;
@@ -2271,17 +2334,24 @@
 
 					vin7csi20: endpoint@0 {
 						reg = <0>;
+						remote-endpoint= <&csi20vin7>;
 					};
 					vin7csi40: endpoint@2 {
 						reg = <2>;
+						remote-endpoint= <&csi40vin7>;
 					};
 				};
 			};
-			/* placeholder */
 		};
 
 		csi20: csi2@fea80000 {
+			compatible = "renesas,r8a774a3-csi2";
 			reg = <0 0xfea80000 0 0x10000>;
+			interrupts = <GIC_SPI 184 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 714>;
+			power-domains = <&sysc R8A774A3_PD_ALWAYS_ON>;
+			resets = <&cpg 714>;
+			status = "disabled";
 
 			ports {
 				#address-cells = <1>;
@@ -2295,35 +2365,48 @@
 
 					csi20vin0: endpoint@0 {
 						reg = <0>;
+						remote-endpoint = <&vin0csi20>;
 					};
 					csi20vin1: endpoint@1 {
 						reg = <1>;
+						remote-endpoint = <&vin1csi20>;
 					};
 					csi20vin2: endpoint@2 {
 						reg = <2>;
+						remote-endpoint = <&vin2csi20>;
 					};
 					csi20vin3: endpoint@3 {
 						reg = <3>;
+						remote-endpoint = <&vin3csi20>;
 					};
 					csi20vin4: endpoint@4 {
 						reg = <4>;
+						remote-endpoint = <&vin4csi20>;
 					};
 					csi20vin5: endpoint@5 {
 						reg = <5>;
+						remote-endpoint = <&vin5csi20>;
 					};
 					csi20vin6: endpoint@6 {
 						reg = <6>;
+						remote-endpoint = <&vin6csi20>;
 					};
 					csi20vin7: endpoint@7 {
 						reg = <7>;
+						remote-endpoint = <&vin7csi20>;
 					};
 				};
 			};
-			/* placeholder */
 		};
 
 		csi40: csi2@feaa0000 {
+			compatible = "renesas,r8a774a3-csi2";
 			reg = <0 0xfeaa0000 0 0x10000>;
+			interrupts = <GIC_SPI 246 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 716>;
+			power-domains = <&sysc R8A774A3_PD_ALWAYS_ON>;
+			resets = <&cpg 716>;
+			status = "disabled";
 
 			ports {
 				#address-cells = <1>;
@@ -2337,31 +2420,38 @@
 
 					csi40vin0: endpoint@0 {
 						reg = <0>;
+						remote-endpoint = <&vin0csi40>;
 					};
 					csi40vin1: endpoint@1 {
 						reg = <1>;
+						remote-endpoint = <&vin1csi40>;
 					};
 					csi40vin2: endpoint@2 {
 						reg = <2>;
+						remote-endpoint = <&vin2csi40>;
 					};
 					csi40vin3: endpoint@3 {
 						reg = <3>;
+						remote-endpoint = <&vin3csi40>;
 					};
 					csi40vin4: endpoint@4 {
 						reg = <4>;
+						remote-endpoint = <&vin4csi40>;
 					};
 					csi40vin5: endpoint@5 {
 						reg = <5>;
+						remote-endpoint = <&vin5csi40>;
 					};
 					csi40vin6: endpoint@6 {
 						reg = <6>;
+						remote-endpoint = <&vin6csi40>;
 					};
 					csi40vin7: endpoint@7 {
 						reg = <7>;
+						remote-endpoint = <&vin7csi40>;
 					};
 				};
 			};
-			/* placeholder */
 		};
 	};
 
-- 
2.7.4

